ndcube/extra_coords/extra_coords.py

```diff
@@ -38,23 +38,30 @@ class ExtraCoordsABC(abc.ABC):
     """
     @abc.abstractmethod
     def add(self,
-            name: str,
+            name: Union[str, Iterable[str]],
             array_dimension: Union[int, Iterable[int]],
             lookup_table: Any,
+            physical_types: Union[str, Iterable[str]] = None,
             **kwargs):
         """
         Add a coordinate to this ``ExtraCoords`` based on a lookup table.
 
         Parameters
         ----------
-        name
-            The name for these world coordinate(s).
-        array_dimension
+        name : `str` or sequence of `str`
+            The name(s) for these world coordinate(s).
+        array_dimension : `int` or `tuple` of `int`
             The pixel dimension(s), in the array, to which this lookup table corresponds.
-        lookup_table
-            The lookup table. Note, if this table is multi-dimensional it must
-            (currently) be specified with its axes in world order, so
-            transposed with respect to the data array.
+        lookup_table : `object` or sequence of `object`
+            The lookup table. A `BaseTableCoordinate <.table_coord>` subclass or anything
+            that can instantiate one, i.e. currently a `~astropy.time.Time`,
+            `~astropy.coordinates.SkyCoord`, or a (sequence of) `~astropy.units.Quantity`.
+            Note, if this table is multi-dimensional it must (currently) be specified with
+            its axes in world order, so transposed with respect to the data array.
```

Comment:
I don't think this is true any more.  I think the axis is now given in array order.

Reply:
So just cut the last sentence, or better explicitly state that it has changed from 1.x?

Reply:
In #416 it sounds as if at this point there is no support for multi-dimensional tables at all.

Reply:
Ah, I slightly misread this sentence initially which has caused a little confusion.  There is support for multi-coordinate tables.  But each individual coordinate must be 1D and can only correspond to one array axis.

On a separate point that caused my initial confusion, the ordering in which the single array axis must be specified is in array order, not pixel order.  For example, say you have a 3-D cube and want to add a 1-D extra coord to the 0th array axis.  In pixel order you would give this as axis 2 because pixel and array ordering are reversed.  However, the API for `ExtraCoords.add` requires you use array ordering to specify the single axis, i.e. 0.  I now realise you were not referring to this part of the API, but my misunderstanding prompted my response.

So to summarise, just cut the last sentence.

Reply:
I have tried not to touch the actual content of the existing description of lookup tables, so did not check if any of that had changed recently. But if that bit is obsolete now I'll cut it.